sale_operating_unit/tests/test_sale_operating_unit.py

```diff
@@ -0,0 +1,163 @@
+        self.sale1 = self._create_sale_order(self.user1.id, self.customer,
+                                             self.product1, self.pricelist,
+                                             self.sale_team_ou1)
+        # Create Sale Order2
+        self.sale2 = self._create_sale_order(self.user2.id, self.customer,
+                                             self.product1, self.pricelist,
+                                             self.sale_team_b2c)
+
+    def _create_user(self, login, groups, company, operating_units):
+        """Create a user."""
+        group_ids = [group.id for group in groups]
+        user = self.res_users_model.create({
+            'name': 'Test Sales User',
+            'login': login,
+            'password': 'demo',
+            'email': 'example@yourcompany.com',
+            'company_id': company.id,
+            'company_ids': [(4, company.id)],
+            'operating_unit_ids': [(4, ou.id) for ou in operating_units],
+            'groups_id': [(6, 0, group_ids)]
+        })
+        return user
+
+    def _create_sale_team(self, uid, operating_unit):
+        """Create a sale team."""
+        team = self.sale_team_model.sudo(uid).with_context(
+            mail_create_nosubscribe=True).create({
+                'name': operating_unit.name,
+                'operating_unit_id': operating_unit.id
+                })
+        return team
+
+    def _create_sale_order(self, uid, customer, product, pricelist, team):
+        """Create a sale order."""
+        sale = self.sale_model.sudo(uid).create({
+            'partner_id': customer.id,
+            'partner_invoice_id': customer.id,
+            'partner_shipping_id': customer.id,
+            'pricelist_id': pricelist.id,
+            'team_id': team.id,
+            'operating_unit_id': team.operating_unit_id.id
+        })
+        self.sale_line_model.sudo(uid).create({
+            'order_id': sale.id,
+            'product_id': product.id,
+            'name': 'Sale Order Line'
+        })
+        return sale
+
+    def _confirm_sale(self, sale):
+        sale.action_confirm()
+        payment = self.payment_model.create({
+            'advance_payment_method': 'all'
+        })
+        sale_context = {
+            'active_id': sale.id,
+            'active_ids': sale.ids,
+            'active_model': 'sale.order',
+            'open_invoices': True,
+        }
+        res = payment.with_context(sale_context).create_invoices()
+        invoice_id = res['res_id']
+        return invoice_id
+
+    def test_security(self):
+        """Test Sale Operating Unit"""
+        # User 2 is only assigned to Operating Unit B2C, and cannot
+        # Access Sales order from Main Operating Unit.
+        sale = self.sale_model.sudo(self.user2.id).search(
+            [('id', '=', self.sale1.id),
+             ('operating_unit_id', '=', self.ou1.id)])
+        self.assertEqual(sale.ids, [], 'User 2 should not have access to '
```

Comment:
`[ ]` blank ??

Reply:
Yes, because the user2 cannot access the sales orders from the main operating unit (the user2 has B2C operating unit). So, the sale order list must be empty.